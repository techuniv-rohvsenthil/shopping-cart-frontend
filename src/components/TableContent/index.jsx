import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const TableContent = (props) => {
  const {item, price, quantity, subtotal} = props;

  return (
    <div className={styles.card} >
          <div className={styles.text}>
            {item}
          </div>
          <div className={styles.text}>
            {price}
          </div>
          <div className={styles.text}>
            {quantity}
          </div>
          <div className={styles.text}>
            {subtotal}
          </div>
      </div>
  );
};


TableContent.propTypes = {
  item: propTypes.string.isRequired,
  price: propTypes.string.isRequired,
  quantity: propTypes.string.isRequired,
  subtotal: propTypes.string.isRequired,
};


export default TableContent;
