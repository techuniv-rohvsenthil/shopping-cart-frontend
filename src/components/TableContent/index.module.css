.card{
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 50rem;
    border-radius: 5%;
    color: black;
    margin-right: 10rem;
    font-family: 'Poppins';
}

.text{
    flex-grow: 1;
    font-size: 1rem;
    padding: 1rem;
    border: solid 0.05rem;
}