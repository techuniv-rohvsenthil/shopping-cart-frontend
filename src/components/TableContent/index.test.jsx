import React from 'react';
import { render } from '@testing-library/react';
import TableContent from './index';

describe('the TableContent component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<TableContent  item="item" price="price" quantity="quantity" subtotal="subtotal"/>);
    expect(asFragment()).toMatchSnapshot();
  });
});