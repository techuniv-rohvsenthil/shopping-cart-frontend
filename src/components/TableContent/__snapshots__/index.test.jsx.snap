// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TableContent component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      item
    </div>
    <div>
      price
    </div>
    <div>
      quantity
    </div>
    <div>
      subtotal
    </div>
  </div>
</DocumentFragment>
`;
